inspirehep/modules/workflows/utils.py

```diff
@@ -164,8 +168,24 @@ def download_file_to_workflow(workflow, name, url):
     through ``ObjectVersion`` and ``FileObject``. This method consumes the
     stream passed to it and saves in its place a ``FileObject`` with the
     details of the downloaded file.
+
+    Consuming the stream might raise a ``ProtocolError`` because the server
+    might terminate the connection before sending any data. In this case we
+    retry ``MAX_RETRIES`` times with exponential backoff before giving up.
     """
-    with closing(requests.get(url=url, stream=True)) as req:
-        if req.status_code == 200:
-            workflow.files[name] = req.raw
-            return workflow.files[name]
+    response = requests.get(url, stream=True)
+    if response.ok:
+        retries = 0
+
+        while retries < MAX_RETRIES:
+            try:
+                workflow.files[name] = response.raw
+                return workflow.files[name]
+            except urllib3.exceptions.ProtocolError:
+                response.close()
+                sleep(2 ** retries)
+                response = requests.get(url, stream=True)
+
+            retries += 1
+
+    raise WorkflowsTaskException()
```

Comment:
This needs an exception message that mentions that there was a problem downloading from `url`, but I'm not sure how to write one that Sentry is able to cluster.